tests/system/providers/slack/example_slack.py

```diff
@@ -0,0 +1,71 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Example Airflow DAG for SlackAPIFileOperator.
+"""
+import os
+from datetime import datetime
+from airflow import models
+from airflow.models.dag import DAG
+from airflow.providers.slack.operators.slack import SlackAPIFileOperator
+
+ENV_ID = os.environ.get("SYSTEM_TESTS_ENV_ID")
+DAG_ID = "slack_example_dag"
+
+with models.DAG(
+    DAG_ID,
+    schedule_interval="@once",
+    start_date=datetime(2021, 1, 1),
+    default_args={'slack_conn_id': 'slack', 'channel': '#general', 'initial_comment': 'Hello World!'},
+    max_active_runs=1,
+    catchup=False,
+    tags=["example", "slack"],
+) as dag:
+
+    # Send file with filename and filetype
+    slack_operator_file = SlackAPIFileOperator(
+        task_id="slack_file_upload_1",
+        filename="/resources/file.txt",
+        filetype="txt",
+    )
+
+    # Send file content
+    slack_operator_file_content = SlackAPIFileOperator(
+        task_id="slack_file_upload_2",
+        content="file content in txt",
+    )
+    # [END slack_operator_howto_guide]
```

Comment:
You'll need to add the corresponding `# [START ...]` tag back and update the [how-to guide](https://github.com/apache/airflow/blob/36a560d51e82fe38ab1ec86cb80cb8d6b4dd245a/docs/apache-airflow-providers-slack/operators/slack_operator_howto_guide.rst) to reflect the new path of the code-snippet locations.

Also, the current example DAG _should probably_ be removed in deference to this system test (especially since it's the same file contents). The AIP mentions that this is "highly recommended" but not required, although, in this case I definitely think it's warranted.

Reply:
added # [START ...] section and updated the [how-to guide](https://github.com/apache/airflow/blob/36a560d51e82fe38ab1ec86cb80cb8d6b4dd245a/docs/apache-airflow-providers-slack/operators/slack_operator_howto_guide.rst) guide.

Reply:
I think @josh-fell is actually right - you should remove the original dag in the same PR - that makes the PR more cherry-pickable - basically the "change" is not split between two chagnes and you can revert it also with a single commit revert.

Reply:
OK. Removed the "legacy" example DAG.